alphalens/utils.py

```diff
@@ -335,6 +352,200 @@ def print_table(table, name=None, fmt=None):
         pd.set_option('display.float_format', prev_option)
 
 
+def get_clean_factor(factor,
+                     forward_returns,
+                     groupby=None,
+                     binning_by_group=False,
+                     quantiles=5,
+                     bins=None,
+                     groupby_labels=None,
+                     max_loss=0.35):
+    """
+    Formats the factor data, forward return data, and group mappings into a
+    DataFrame that contains aligned MultiIndex indices of timestamp and asset.
+    The returned data will be formatted to be suitable for Alphalens functions.
+
+    It is safe to skip a call to this function and still make use of Alphalens
+    functionalities as long as the factor data conforms to the format returned
+    from get_clean_factor_and_forward_returns and documented here
+
+    Parameters
+    ----------
+    factor : pd.Series - MultiIndex
+        A MultiIndex Series indexed by timestamp (level 0) and asset
+        (level 1), containing the values for a single alpha factor.
+        ::
+            -----------------------------------
+                date    |    asset   |
+            -----------------------------------
+                        |   AAPL     |   0.5
+                        -----------------------
+                        |   BA       |  -1.1
+                        -----------------------
+            2014-01-01  |   CMG      |   1.7
+                        -----------------------
+                        |   DAL      |  -0.1
+                        -----------------------
+                        |   LULU     |   2.7
+                        -----------------------
+
+    forward_returns : pd.DataFrame - MultiIndex
+        Forward returns indexed by date and asset.
+        ::
```

Comment:
Since this is a top level API it would be better to improve the forward_returns description,
something like this:

A MultiIndex DataFrame indexed by timestamp (level 0) and asset
(level 1), containing the forward returns for assets.
Forward returns column names must follow the format accepted by
pd.Timedelta (e.g. '1D', '30m', '3h15m', '1D1h', etc).
'date' index freq property must be set to a trading calendar (pandas
DateOffset), see infer_trading_calendar for more details. This information is
currently used only in cumulative returns computation